util/EFSOI_Utilities/src/gridinfo_efsoi.f90

```diff
@@ -0,0 +1,461 @@
+! read latitudes, longitudes and pressures for analysis grid,
+! broadcast to each task.
+use sigio_module, only: sigio_head, sigio_data, sigio_sclose, sigio_sropen, &
+                        sigio_srohdc, sigio_sclose, sigio_srhead, sigio_axdata
+			
+use nemsio_module, only: nemsio_gfile,nemsio_open,nemsio_close,&
+                         nemsio_getfilehead,nemsio_getheadvar,&
+                         nemsio_readrecv,nemsio_init, nemsio_realkind
+						 
+use module_fv3gfs_ncio, only: Dataset, Variable, Dimension, open_dataset,&
+                              read_attribute, close_dataset, get_dim, read_vardata
+
+
+implicit none
+
+type(Dataset) :: dset
+type(Dimension) :: londim,latdim,levdim
+character(len=120), intent(in) :: fileprefix
+logical, intent(in)            :: reducedgrid
+
+integer(i_kind) nlevsin, ierr, iunit, k, nn, idvc 
+character(len=500) :: filename
+real(r_kind), allocatable, dimension(:) :: ak,bk,spressmn,tmpspec
+real(r_kind), allocatable, dimension(:,:) :: pressimn,presslmn,values_2d
+real(r_single),allocatable,dimension(:,:,:) :: nems_vcoord
+real(r_kind) kap,kapr,kap1
+real(nemsio_realkind), dimension(nlons*nlats) :: nems_wrk
+type(sigio_data) sigdata
+type(sigio_head) sighead
+type(nemsio_gfile) :: gfile
+
+integer(i_kind) iret,i,j,nlonsin,nlatsin,u_ind,v_ind,tv_ind,q_ind,ps_ind
+integer, intent(in) :: nc3d
+integer, dimension(0:nc3d), intent(in) :: clevels
+
+
+iunit = 77
+kap = rd/cp
+kapr = cp/rd
+kap1 = kap + one
+nlevs_pres=nlevs+1
+
+
+if (nproc .eq. 0) then
+   filename = trim(adjustl(datapath))//trim(adjustl(fileprefix))//"ensmean"
+   print *,'reading file: ',trim(filename)
+   
+   if (use_gfs_nemsio) then
+       call nemsio_init(iret=iret)
+       if(iret/=0) then
+          write(6,*)'grdinfo: gfs model: problem with nemsio_init, iret=',iret, ', file: ', trim(filename)
+          call stop2(23)
+       end if
+       call nemsio_open(gfile,filename,'READ',iret=iret)
+       if (iret/=0) then
+          write(6,*)'grdinfo: gfs model: problem with nemsio_open, iret=',iret, ', file: ', trim(filename)
+          call stop2(23)
+       endif
+       call nemsio_getfilehead(gfile,iret=iret, dimx=nlonsin, dimy=nlatsin,&
+                               dimz=nlevsin,jcap=ntrunc,idvc=idvc)
+       ! set ntrunc to nlats if missing
+       ! (only used for inflation smoothing and mass balance adjustment if use_gfsnemsio = T)
+       ! FV3GFS write component does not include JCAP, infer from nlatsin
+       if (ntrunc < 0) ntrunc = nlatsin-2
+       if (iret/=0) then
+          write(6,*)'grdinfo: gfs model: problem with nemsio_getfilehead, iret=',iret, ', file: ', trim(filename)
+          call stop2(23)
+       endif
+       print *,'ntrunc = ',ntrunc
+       if (nlons /= nlonsin .or. nlats /= nlatsin .or. nlevs /= nlevsin) then
+         print *,'incorrect dims in nemsio file'
+         print *,'expected',nlons,nlats,nlevs
+         print *,'got',nlonsin,nlatsin,nlevsin
+         call stop2(23)
+       end if
```

Comment:
Do we need to keep capability for nemsio in FSOI computations?
@dtkleist 
If not, we should remove this option and simplify the duplications

Reply:
I don't see why we would. Let's move forward with netcdf / v16 style.

Reply:
works for me

Reply:
I've removed gridinfo_efsoi.f90, now using regular enkf version